hail/src/main/scala/is/hail/expr/ir/LiftLiterals.scala

```diff
@@ -77,11 +77,11 @@ object LiftLiterals {
 
   def apply(ir: BaseIR): BaseIR = {
     MapIR.mapBaseIR(ir, {
-      case MatrixMapRows(child, newRow, newKey) =>
+      case MatrixMapRows(child, newRow) =>
         val literals = getLiterals(newRow)
         val rewriteChild = addLiterals(child, literals)
         removeLiterals(
-          MatrixMapRows(rewriteChild, rewriteIR(newRow, rewriteChild.typ.globalType, literals), newKey),
+          MatrixMapRows(rewriteChild, rewriteIR(newRow, rewriteChild.typ.globalType, literals)),
           literals)
       case MatrixMapEntries(child, newEntries) =>
```

Comment:
Does MatrixKeyRowsBy need a case here? Or is it no because you're not adding new fields?

Reply:
the things that need to appear in LiftLiterals are methods that (a) contain Value IRs, and (b) call Compile on those IRs. MatrixKeyRowsBy doesn't, so it doesn't need to appear here